src/Controller/ProductController.php

```diff
@@ -24,26 +25,33 @@ public function __construct(EntityManagerInterface $entityManager)
     public function index(Request $request): Response
     {
         $categories = $this->entityManager->getRepository(Category::class)->findAll();
-        $mocktails = $this->entityManager->getRepository(Product::class)->findAll();
+        $products = $this->entityManager->getRepository(Product::class)->findAll();
 
-        $mocktailsPerPage = 10;
-        $totalMocktails = count($mocktails);
-        $pagesNumber = ceil($totalMocktails / $mocktailsPerPage);
+        $productsPerPage = 10;
+        $totalproducts = count($products);
+        $pagesNumber = ceil($totalproducts / $productsPerPage);
 
         $page = $request->query->getInt('page', 1);
 
-        return $this->render('mocktail/index.html.twig', [
+        return $this->render('product/index.html.twig', [
             'pageName' => "Nos Mocktails",
             'categories' => $categories,
-            'mocktails' => $mocktails,
+            'products' => $products,
             'pagesNumber' => $pagesNumber,
             'currentPage' => $page,
         ]);
     }
 
-    #[Route('/show', name: 'show')]
-    public function show(): Response
+    #[Route('/{id}', name: 'show', requirements: ['id' => '\d+'])]
+    public function show(int $id, ProductRepository $productRepository): Response
```

Comment:
Use param converter, Product $product instead of ProductRepository then you won't even have to fetch it. Symfony will make it for you